File: leetcodes/206. Reverse Linked List/reverseLinkedList.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ReverseList(ListNode head) {
      //create two pointer
      if(head == null || head.next == null) return head;
      ListNode cur = head;
      ListNode pre = null;
      while( cur != null) {
        ListNode temp = cur.next;
        cur.next = pre;
        pre = cur;
        cur = temp;
      }
      return pre;
    }
}

File: leetcodes/206. Reverse Linked List/reverseLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prev = None
        tempNext = None
        cur = head
        while(cur != None):
            tempNext = cur.next
            cur.next = prev
            prev = cur
            cur = tempNext
        return prev       

            